libs/ledgerjs/packages/swift-bridge-hw-transport-ble/src/TransportBLESwift.ts

```diff
@@ -0,0 +1,52 @@
+import Transport from "@ledgerhq/hw-transport";
+
+/**
+ * iOS Bluetooth Transport implementation
+ */
+export default class TransportBLESwift extends Transport {
+  /**
+   * SwiftTransport lives in the `global` scope and gets injected by the Swift Native Module
+   */
+  private transport: SwiftTransportType;
+
+  constructor() {
+    super();
+    this.transport = SwiftTransport.create();
+  }
+
+  /**
+   * communicate with a BLE transport
+   */
+  async exchange(apdu: Buffer): Promise<Buffer> {
+    const response = await this.promisify(this.transport.exchange(apdu));
+    return Buffer.from(response);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  promisify(callback: Callback): Promise<any> {
+    return new Promise((resolve, reject) => {
+      callback(function (response, error) {
+        if (error == "") {
+          resolve(response);
+        } else {
+          reject(error);
+        }
```

Comment:
This feels a bit counterintuitive 🤔 

What is the reason to compare specifically `error == ""`?

Is it similar to this suggestion, or is there some specific cases where response can be nullish but we still want to resolve?

```suggestion
        if (response) {
          resolve(response);
        } else {
          reject(error);
        }
```

Reply:
I'm not sure why this was written this way originally, but after testing this, checking for a response instead seems to be fine.